staging/src/k8s.io/client-go/rest/transport.go

```diff
@@ -111,5 +82,29 @@ func (c *Config) TransportConfig() (*transport.Config, error) {
 			Extra:    c.Impersonate.Extra,
 		},
 		Dial: c.Dial,
-	}, nil
+	}
+	if c.ExecProvider != nil {
+		provider, err := exec.GetAuthenticator(c.ExecProvider)
+		if err != nil {
+			return nil, err
+		}
+		if err := provider.UpdateTransportConfig(conf); err != nil {
+			return nil, err
+		}
+	}
+	if c.AuthProvider != nil {
```

Comment:
@ericchiang remind me the use case for setting both AuthProvider and ExecProvider? was there a reason we didn't make those exclusive when adding ExecProvider in 1.10?

Reply:
If they're not mutually exclusive that wasn't intended. I wouldn't expect you to be able to provide AuthProvider and ExecProvider.

Maybe I should change that as part of the beta promotion?

Reply:
I agree, this should be disallowed as part of beta promotion PR